src/aggregator/client/queue.go

```diff
@@ -110,20 +110,25 @@ type writeFn func([]byte) error
 type queue struct {
 	sync.RWMutex
 
-	log      *zap.Logger
-	metrics  queueMetrics
-	dropType DropType
-	instance placement.Instance
-	conn     *connection
-	bufCh    chan protobuf.Buffer
-	doneCh   chan struct{}
-	closed   bool
-	wg       sync.WaitGroup
+	log           *zap.Logger
+	metrics       queueMetrics
+	dropType      DropType
+	instance      placement.Instance
+	conn          *connection
+	bufCh         chan protobuf.Buffer
+	doneCh        chan struct{}
+	closed        bool
+	batchSize     int
+	buf           []byte
+	writeInterval time.Duration
+	wg            sync.WaitGroup
 
 	writeFn writeFn
 }
 
 func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
+	const batchSize = 16384
+	const minWriteInterval = 250 * time.Millisecond
```

Comment:
super nits:
- 16384 can become: `batchSize = 2 << 13 // 16k`
- combine into single `const` stanza
```go
const (
  batchSize = 2 << 13 // 16k
  minWriteInterval = 250 * time.Millisecond
)
```

Reply:
Also, I'd feel more comfortable with 100ms than 250ms, do you think it will make a huge difference?  Any level of buffering will be big.